vault/expiration.go

```diff
@@ -137,7 +137,7 @@ type ExpirationManager struct {
 	quitCh             chan struct{}
 
 	// do not hold coreStateLock in any API handler code - it is already held
-	coreStateLock     *DeadlockRWMutex
+	coreStateLock     *RWMutex
```

Comment:
It's a little weird to use a pointer to an interface, since that's already a pointer.  You may have good reasons, in which case maybe a comment is called for?

Reply:
I did it this way to [leave the existing expiration manager instantiation unchanged](https://github.com/hashicorp/vault/blob/v1.11.3/vault/expiration.go#L345). I'm not sure what would be informative to add as a comment but open to suggestions (or pointers - _zing!_ - on better implementation ideas)

Reply:
Well, I was basically saying: can you explain why you're doing this, and if it's a good reason, we should add a comment to the code, and if we find we don't need to do this, even better.

Anyway, if coreStateLock becomes an interface, the instantiation can be basically unchanged, except syntactically (i.e. removal of the ampersand).